File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        ans = 0
        
        def dfs(node, prevMax):
            nonlocal ans

            if not node:
                return

            if node.val >= prevMax:
                ans += 1

            dfs(node.left, max(node.val, prevMax))
            dfs(node.right, max(node.val, prevMax))

        dfs(root, float("-inf"))

        return ans

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[][]}
     */
    levelOrder(root) {
        let res = []
        let queue = []

        if (root) queue.push(root)

        while (queue.length) {
           let level = []
           for (let i = queue.length; i > 0; i--) {
           
            const node = queue.shift()

            if (node.left) queue.push(node.left)
            if (node.right) queue.push(node.right)

            level.push(node.val)
           } 
           res.push(level)
        }

        return res
    }
}
